src/conq/navigation_lib/openai_nav/openai_nav_client.py

```diff
@@ -0,0 +1,96 @@
+import ast
+import os
+from typing import List
+
+from dotenv import load_dotenv
+from openai import OpenAI
+from ratelimit import limits
+
+from conq.api_keys import OPENAI_API_KEY
+
+# Change this based on which API you are using
+os.environ["OPENAI_API_KEY"] = OPENAI_API_KEY.get()
+
+ONE_MINUTE = 60  # Constant for seconds in a minute
+
+
+class OpenAINavClient:
+    def __init__(self, locations: List[str]):
+        # Sets the current working directory to be the same as the file.
+        os.chdir(os.path.dirname(os.path.abspath(__file__)))
+
+        try:
+            if load_dotenv(".env") is False:
+                raise TypeError
+        except TypeError:
+            print("Unable to load .env file.")
+            quit()
+
+        # Create Azure client
+        # self.llm_client = AzureOpenAI(
+        #     api_key=os.environ["OPENAI_API_KEY"],
+        #     api_version=os.environ["API_VERSION"],
+        #     azure_endpoint=os.environ["openai_api_base"],
+        #     organization=os.environ["OPENAI_organization"],
+        # )
+        self.llm_client = OpenAI(
+            api_key=os.environ["OPENAI_API_KEY"],
+        )
+
+        # Locations in our environment
+        self.locations = locations
+
+        # System context to give to the LLM before asking question
+        self.system_context = "Assistant is a large language model trained by OpenAI."
+
+    @limits(calls=3, period=ONE_MINUTE)
+    def find_probable_location_for_object(self, object: str):
+        # Build the query string
+        query = "I observe the following structures while exploring a small-scale farm:"
+
+        for index in range(len(self.locations)):
+            query += f"\n{index + 1}. {self.locations[index]}"
+
+        query += f"\n\nPlease rank the structure based on how likely I am to find a {object} in them. Please provide the response in a plain text string representing a python dictionary, {{\"name\": probability}} and only include the location with whitespace connected with _ as the keys and probabilities as the values"
```

Comment:
Ahhh the age-old problem of multi-line strings in Python... A pain due to Python's enforced whitespace. So I'm a bit of a stickler typically about line length and my cutoff point is typically 100 characters. Can I suggest you change this query formation to use the `str.join()` method?

```python
query_list = ["I observe the following structures while exploring a small-scale farm:"]
query_list.extend([f"{index + 1}. {location}" for index, location in enumerate(self.locations)])
query_list.extend([
    f"\nPlease rank the structure based on how likely I am to find a {object} in them. Please "
    "provide the response in a plain text string representing a python dictionary, "
    f"{{\"name\": probability}} and only include the location with whitespace connected with _ as "
    "the keys and probabilities as the values"
])
query = "\n".join(query_list)
```

Let me know if you don't like this method and I can suggest a different way of doing this that doesn't require the `.join()` method

Reply:
This is soooo much cleaner! Done in [624417e](https://github.com/UM-ARM-Lab/conq_python/pull/52/commits/624417e6bfac29a0d92bf839e2457c609ebf63b4)

Reply:
Great!